internal/credential/vault/repository_credentials.go

```diff
@@ -54,8 +87,28 @@ func (r *Repository) Issue(ctx context.Context, sessionId string, requests []cre
 		if minLease > cred.getExpiration() {
 			minLease = cred.getExpiration()
 		}
-		insertQuery, insertQueryValues := cred.insertQuery()
-		updateQuery, updateQueryValues := cred.updateSessionQuery(lib.Purpose)
+
+		var underlyingCred *Credential
+		switch c := cred.(type) {
+		case *usrPassCred:
+			underlyingCred = c.Credential
+		case *sshPrivateKeyCred:
+			underlyingCred = c.Credential
+		case *baseCred:
+			underlyingCred = c.Credential
+		default:
+			return nil, errors.New(ctx, errors.InvalidDynamicCredential, op, fmt.Sprintf("vault dynamicCred (typeof %T) could not be mapped back to vault.Credential", c))
+		}
```

Comment:
I think we can eliminate the need for this type switch by expanding the `dynamicCred` interface. There are a few ways this could be done. The main motivation would be so that when new implementations of `dynamicCred` are added, we won't need to remember to add a case to this switch. Instead if the interface isn't implemented completely, we would get a compile-time error.

One way to do this would be to update `dynamicCred` to have:

```go
type dynamicCred interface {
  // other methods
  getCredential() *Credential
}
```

Then update `baseCred` to implement this method:

```go
func (bc *baseCred) getCredential() *Credential    { return bc.Credential }
```

Then this switch can be replaced with:

```go
underlyingCred := cred.getCredential()
```